src/main/java/io/apimatic/core/ResponseHandler.java

```diff
@@ -202,13 +204,28 @@ private void validateResponse(Context httpContext) throws ExceptionType {
         Response response = httpContext.getResponse();
         int statusCode = response.getStatusCode();
         String errorCode = String.valueOf(statusCode);
+        String defaultErrorCode = "";
+        Matcher match = Pattern.compile("^[(4|5)[0-9]]{3}").matcher(errorCode);
+        if (match.find()) {
+            defaultErrorCode = errorCode.charAt(0) + "XX";
+        }
 
-        if (localErrorCases != null && localErrorCases.containsKey(errorCode)) {
-            localErrorCases.get(errorCode).throwException(httpContext);
+        if (localErrorCases != null) {
```

Comment:
This piece of code is also repeating for globalErrorCases, cosider extracting it as a method, which return early it the hashMap is null

Reply:
Done